Play Services Crash Report

```
type: crash
osVersion: google/raven/raven:13/TQ1A.230105.002/2023011000:user/release-keys
package: com.google.android.gms:225014044
process: com.google.android.gms.persistent
processUptime: 29375904 + 272 ms
GmsCompatConfig version: 25

java.lang.RuntimeException: Unable to create service com.google.android.gms.nearby.discovery.service.DiscoveryService: java.lang.NullPointerException: Attempt to read from field 'android.os.IInterface com.android.server.notification.ManagedServices$ManagedServiceInfo.service' on a null object reference in method 'com.android.server.notification.ManagedServices$ManagedServiceInfo com.android.server.notification.ManagedServices.getServiceFromTokenLocked(android.os.IInterface)'
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4504)
	at android.app.ActivityThread.-$$Nest$mhandleCreateService(Unknown Source:0)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2162)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7878)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ExecInit.main(ExecInit.java:49)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:355)
Caused by: java.lang.NullPointerException: Attempt to read from field 'android.os.IInterface com.android.server.notification.ManagedServices$ManagedServiceInfo.service' on a null object reference in method 'com.android.server.notification.ManagedServices$ManagedServiceInfo com.android.server.notification.ManagedServices.getServiceFromTokenLocked(android.os.IInterface)'
	at android.os.Parcel.createExceptionOrNull(Parcel.java:3027)
	at android.os.Parcel.createException(Parcel.java:3005)
	at android.os.Parcel.readException(Parcel.java:2981)
	at android.os.Parcel.readException(Parcel.java:2923)
	at android.app.INotificationManager$Stub$Proxy.createNotificationChannelGroups(INotificationManager.java:3481)
	at android.app.NotificationManager.createNotificationChannelGroups(NotificationManager.java:886)
	at android.app.NotificationManager.createNotificationChannelGroup(NotificationManager.java:875)
	at aaal.n(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at aaam.n(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at aaax.n(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at axbf.f(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at aysi.i(:com.google.android.gms@225014044@22.50.14 (190400-499278674):3)
	at aysi.<init>(:com.google.android.gms@225014044@22.50.14 (190400-499278674):2)
	at axth.a(:com.google.android.gms@225014044@22.50.14 (190400-499278674):9)
	at axbw.i(:com.google.android.gms@225014044@22.50.14 (190400-499278674):3)
	at axbw.d(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at axbw.b(:com.google.android.gms@225014044@22.50.14 (190400-499278674):1)
	at axbw.c(:com.google.android.gms@225014044@22.50.14 (190400-499278674):3)
	at aypu.<init>(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at axth.a(:com.google.android.gms@225014044@22.50.14 (190400-499278674):11)
	at axbw.i(:com.google.android.gms@225014044@22.50.14 (190400-499278674):3)
	at axbw.d(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at axbw.b(:com.google.android.gms@225014044@22.50.14 (190400-499278674):1)
	at axbw.c(:com.google.android.gms@225014044@22.50.14 (190400-499278674):3)
	at axtp.<init>(:com.google.android.gms@225014044@22.50.14 (190400-499278674):2)
	at axth.a(:com.google.android.gms@225014044@22.50.14 (190400-499278674):1)
	at axbw.i(:com.google.android.gms@225014044@22.50.14 (190400-499278674):3)
	at axbw.d(:com.google.android.gms@225014044@22.50.14 (190400-499278674):0)
	at axbw.b(:com.google.android.gms@225014044@22.50.14 (190400-499278674):1)
	at axbw.c(:com.google.android.gms@225014044@22.50.14 (190400-499278674):3)
	at com.google.android.gms.nearby.discovery.service.DiscoveryChimeraService.onCreate(:com.google.android.gms@225014044@22.50.14 (190400-499278674):4)
	at gss.onCreate(:com.google.android.gms@225014044@22.50.14 (190400-499278674):4)
	at xpz.onCreate(:com.google.android.gms@225014044@22.50.14 (190400-499278674):4)
	at android.app.ActivityThread.handleCreateService(ActivityThread.java:4491)
	... 11 more

```

--- a/services/core/java/com/android/server/notification/ManagedServices.java
+++ b/services/core/java/com/android/server/notification/ManagedServices.java
@@ -1002,6 +1002,12 @@ abstract public class ManagedServices {
         return null;
     }
 
+    protected boolean isServiceTokenValid(IInterface service) {
+        synchronized (mMutex) {
+            return isServiceTokenValidLocked(service);
+        }
+    }
+
     protected boolean isServiceTokenValidLocked(IInterface service) {
         if (service == null) {
             return false;

--- a/services/core/java/com/android/server/notification/NotificationManagerService.java
+++ b/services/core/java/com/android/server/notification/NotificationManagerService.java
@@ -10390,7 +10390,7 @@ public class NotificationManagerService extends SystemService {
      * given NAS is bound in.
      */
     private boolean isInteractionVisibleToListener(ManagedServiceInfo info, int userId) {
-        boolean isAssistantService = mAssistants.isServiceTokenValidLocked(info.service);
+        boolean isAssistantService = mAssistants.isServiceTokenValid(info.service);
         return !isAssistantService || info.isSameUser(userId);
     }
 
@@ -11651,7 +11651,7 @@ public class NotificationManagerService extends SystemService {
                 BackgroundThread.getHandler().post(() -> {
                     if (info.isSystem
                             || hasCompanionDevice(info)
-                            || mAssistants.isServiceTokenValidLocked(info.service)) {
+                            || mAssistants.isServiceTokenValid(info.service)) {
                         notifyNotificationChannelChanged(
                                 info, pkg, user, channel, modificationType);
                     }

